plugins/view-transitions/includes/theme.php

```diff
@@ -306,6 +307,7 @@ function plvt_load_view_transitions(): void {
 	$default_animation_args       = isset( $theme_support['default-animation-args'] ) ? (array) $theme_support['default-animation-args'] : array();
 	$default_animation_stylesheet = $animation_registry->get_animation_stylesheet( $theme_support['default-animation'], $default_animation_args );
 	if ( '' !== $default_animation_stylesheet ) {
+		$default_animation_stylesheet = plvt_replace_animation_duration( $default_animation_stylesheet, absint( $theme_support['default-animation-duration'] ) );
 		wp_add_inline_style( 'plvt-view-transitions', $default_animation_stylesheet );
 	}
```

Comment:
I think at the moment this PR is missing to cater for the default "fade" animation, which doesn't use a stylesheet (since it's the browser default).

So I think you'll need to add an `else` here that manually injects the CSS custom property and a `::view-transition-group(*)` rule with the `animation-duration` CSS property. That way, the transition animation duration setting can also control the length of the default animation.

Reply:
+1